### PR TITLE
Implement continuous query backfill

### PR DESCRIPTION
**Pull request by DanielMorsing**
The syntax for this feature is `BACKFILL cqname ON database UNTIL
timeliteral`

This will fill in the continuous query from now until the time
literal. The time literal can be an expression such as `now() - 10d`

### CODE REVIEW

**Comment by pauldix** [comment]
We should be able to backfill specific ranges of time too

File: influxql/ast.go [hunk 1]
```diff
@@ -552,6 +554,31 @@ func (s *SetPasswordUserStatement) RequiredPrivileges() ExecutionPrivileges {
...
+
+func (b *BackfillStatement) RequiredPrivileges() ExecutionPrivileges {
+	return []ExecutionPrivilege{
+		{
+			Admin:     false,
+			Name:      b.Database,
+			Privilege: WritePrivilege,
+		},
+	}
+}
```
Comment (dgnorton) [inline influxql/ast.go]: Wondering if the required privileges should come from the CQ being run.

File: influxql/parser.go [hunk 2]
```diff
@@ -265,6 +267,59 @@ func (p *Parser) parseSetPasswordUserStatement() (*SetPasswordUserStatement, err
...
+	}
+	stmt.From = expr
+
+	// Get the UNTIL keyword. This is optional, so if we see EOF, just ignore it.
+	tok, pos, lit = p.scanIgnoreWhitespace()
+	if tok == EOF {
+		return stmt, nil
+	} else if tok != UNTIL {
+		return nil, newParseError(tokstr(tok, lit), []string{"UNTIL"}, pos)
+	}
```
Comment (dgnorton) [inline influxql/parser.go]: If it's not an `UNTIL` token, it should unscan it and return.  The main loop should handle whatever comes next (could be `EOF` or `SEMICOLON`).

File: influxql/ast.go (hunk 1 above)
Comment (DanielMorsing) [inline influxql/ast.go]: Problem with that is that we only have the CQ name at this point, not the CQ contents. In order to use the CQ being run, we need to look up the contents, which we don't have any handles to do at the moment.

File: influxql/parser.go [hunk 3]
```diff
@@ -265,6 +267,58 @@ func (p *Parser) parseSetPasswordUserStatement() (*SetPasswordUserStatement, err
...
+
+	expr, err := p.ParseExpr()
+	if err != nil {
+		return nil, err
+	}
+	stmt.From = expr
+
+	// Get the UNTIL keyword. This is optional, so if we see EOF, just ignore it.
+	tok, pos, lit = p.scanIgnoreWhitespace()
+	if tok != UNTIL {
```
Comment (otoolep) [inline influxql/parser.go]: If the token is something else other than EOF, isn't that an error?

File: influxql/parser_test.go [hunk 4]
```diff
@@ -1457,10 +1457,10 @@ func TestParser_ParseStatement(t *testing.T) {
 		},
 
 		// Errors
-		{s: ``, err: `found EOF, expected SELECT, DELETE, SHOW, CREATE, DROP, GRANT, REVOKE, ALTER, SET at line 1, char 1`},
+		{s: ``, err: `found EOF, expected SELECT, DELETE, SHOW, CREATE, DROP, GRANT, REVOKE, ALTER, SET, BACKFILL at line 1, char 1`},
 		{s: `SELECT`, err: `found EOF, expected identifier, string, number, bool at line 1, char 8`},
 		{s: `SELECT time FROM myseries`, err: `at least 1 non-time field must be queried`},
-		{s: `blah blah`, err: `found blah, expected SELECT, DELETE, SHOW, CREATE, DROP, GRANT, REVOKE, ALTER, SET at line 1, char 1`},
+		{s: `blah blah`, err: `found blah, expected SELECT, DELETE, SHOW, CREATE, DROP, GRANT, REVOKE, ALTER, SET, BACKFILL at line 1, char 1`},
```
Comment (otoolep) [inline influxql/parser_test.go]: Shouldn't there be some `BACKFILL` tests here?

**Comment by otoolep** [comment]
Generally makes sense, but I have some questions about the parsing.

File: influxql/parser.go (hunk 3 above)
Comment (dgnorton) [inline influxql/parser.go]: Multiple statements could be separated by a semicolon.

**Pull request closed without merging**